File: src/frontend/src/types/project/projectLayer.js
export class ProjectLayer {
  constructor(data) {
    this.layerId = data?.layerId
    this.layerName = data?.layerName
    this.width = data?.width
    this.height = data?.height
    this.image = data?.image
    this.i2i = data?.i2i
    this.mask = data?.mask
  }
}

File: src/backend/yt_diffuser/types/project/project_layer.py
from typing import Union, Optional
import uuid

from pydantic import BaseModel, Field, UUID4

class ProjectLayer(BaseModel):
    """
    プロジェクトのレイヤー情報を保持するクラス
    """

    layer_id: Union[UUID4, str] = Field(default_factory=uuid.uuid4)
    layer_name: str

    image: Optional[str] = None
    i2i: Optional[str] = None
    mask: Optional[str] = None